src/java/picard/fingerprint/FingerprintChecker.java

```diff
@@ -586,23 +582,32 @@ public static MatchResults calculateMatchResults(final Fingerprint observedFp, f
             final HaplotypeProbabilities probs1 = observedFp.get(haplotypeBlock);
             if (probs1 == null) continue;
 
-            final HaplotypeProbabilityOfNormalGivenTumor normalizedProbs1 = new HaplotypeProbabilityOfNormalGivenTumor(probs1, pLoH);
-            final HaplotypeProbabilityOfNormalGivenTumor normalizedProbs2 = new HaplotypeProbabilityOfNormalGivenTumor(probs2, pLoH);
+            final HaplotypeProbabilityOfNormalGivenTumor prob1AssumingDataFromTumor = new HaplotypeProbabilityOfNormalGivenTumor(probs1, pLoH);
+            final HaplotypeProbabilityOfNormalGivenTumor prob2AssumingDataFromTumor = new HaplotypeProbabilityOfNormalGivenTumor(probs2, pLoH);
 
             // If one is from genotype data we'd like to report the output relative
             // to the genotyped SNP instead of against a random SNP from the haplotype
             final Snp snp = probs2.getRepresentativeSnp();
             final DiploidGenotype externalGenotype = probs2.getMostLikelyGenotype(snp);
             final LocusResult lr = new LocusResult(snp,
-                                                   externalGenotype,
-                                                   probs1.getMostLikelyGenotype(snp),
-                                                   probs1.getObsAllele1(),
-                                                   probs1.getObsAllele2(),
-                                                   probs1.getLodMostProbableGenotype(),
-                                                   probs1.shiftedLogEvidenceProbabilityGivenOtherEvidence(probs2),
+                    externalGenotype,
+                    probs1.getMostLikelyGenotype(snp),
+                    probs1.getObsAllele1(),
+                    probs1.getObsAllele2(),
+                    probs1.getLodMostProbableGenotype(),
+                    // expected sample log-likelihood
+                    probs1.shiftedLogEvidenceProbabilityGivenOtherEvidence(probs2),
+                    // random sample log-likelihood
                     probs1.shiftedLogEvidenceProbability(),
-                    probs2.shiftedLogEvidenceProbabilityGivenOtherEvidence(normalizedProbs1)-probs2.shiftedLogEvidenceProbability(),
-                    probs1.shiftedLogEvidenceProbabilityGivenOtherEvidence(normalizedProbs2)-probs1.shiftedLogEvidenceProbability());
+
+            // probs1 is tumor probs2 is normal, correct sample lod
+            prob1AssumingDataFromTumor.shiftedLogEvidenceProbabilityGivenOtherEvidence(probs2) -
```

Comment:
this line (604) looks like a change, but it actually isn't. happy to explain with a whiteboard.